internal/storage/v2/clickhouse/tracestore/dbmodel/to_dbmodel.go

```diff
@@ -0,0 +1,354 @@
+			Data: pair.valueCol,
+		})
+	}
+	return result
+}
+
+const (
+	ValueTypeBool   = pcommon.ValueTypeBool
+	ValueTypeDouble = pcommon.ValueTypeDouble
+	ValueTypeInt    = pcommon.ValueTypeInt
+	ValueTypeStr    = pcommon.ValueTypeStr
+	ValueTypeBytes  = pcommon.ValueTypeBytes
+)
+
+// AppendNestedAttributeGroupToColumns Writes a complete set of pcommon.Map to the database. NestedAttributesGroup and pcommon.Map have a one-to-many relationship.
+func appendNestedAttributeGroupToColumns(attributeType AttributeType, nestedGroup NestedAttributesGroup) {
+	var boolKeys [][]string
+	var boolValues [][]bool
+	var doubleKeys [][]string
+	var doubleValues [][]float64
+	var intKeys [][]string
+	var intValues [][]int64
+	var strKeys [][]string
+	var strValues [][]string
+	var bytesKeys [][]string
+	var bytesValues [][]string
+	for _, group := range nestedGroup.AttributesGroups {
+		boolKeys = append(boolKeys, group.BoolKeys)
+		boolValues = append(boolValues, group.BoolValues)
+		doubleKeys = append(doubleKeys, group.DoubleKeys)
+		doubleValues = append(doubleValues, group.DoubleValues)
+		intKeys = append(intKeys, group.IntKeys)
+		intValues = append(intValues, group.IntValues)
+		strKeys = append(strKeys, group.StrKeys)
+		strValues = append(strValues, group.StrValues)
+		bytesKeys = append(bytesKeys, group.BytesKeys)
+		bytesValues = append(bytesValues, group.BytesValues)
+	}
+
+	attrsColumnsMap := AllAttributeColumns[attributeType]
+
+	boolKeyCol := attrsColumnsMap[ValueTypeBool].keyCol
+	boolKeyCol.(*proto.ColArr[[]string]).Append(boolKeys)
+	boolValueCol := attrsColumnsMap[ValueTypeBool].valueCol
+	boolValueCol.(*proto.ColArr[[]bool]).Append(boolValues)
+
+	doubleKeyCol := attrsColumnsMap[ValueTypeDouble].keyCol
+	doubleKeyCol.(*proto.ColArr[[]string]).Append(doubleKeys)
+	doubleValueCol := attrsColumnsMap[ValueTypeDouble].valueCol
+	doubleValueCol.(*proto.ColArr[[]float64]).Append(doubleValues)
+
+	intKeyCol := attrsColumnsMap[ValueTypeInt].keyCol
+	intKeyCol.(*proto.ColArr[[]string]).Append(intKeys)
+	intValueCol := attrsColumnsMap[ValueTypeInt].valueCol
+	intValueCol.(*proto.ColArr[[]int64]).Append(intValues)
+
+	strKeyCol := attrsColumnsMap[ValueTypeStr].keyCol
+	strKeyCol.(*proto.ColArr[[]string]).Append(strKeys)
+	strValueCol := attrsColumnsMap[ValueTypeStr].valueCol
+	strValueCol.(*proto.ColArr[[]string]).Append(strValues)
+
+	bytesKeyCol := attrsColumnsMap[ValueTypeBytes].keyCol
+	bytesKeyCol.(*proto.ColArr[[]string]).Append(bytesKeys)
+	bytesValueCol := attrsColumnsMap[ValueTypeBytes].valueCol
+	bytesValueCol.(*proto.ColArr[[]string]).Append(bytesValues)
+}
+
+// AppendGroup Writes a complete pcommon.Map to the database. AttributesGroup and pcommon.Map have a one-to-one relationship.
+func appendAttributeGroupToColumns(attributeType AttributeType, group AttributesGroup) {
+	attrColumnsMap := AllAttributeColumns[attributeType]
+
+	boolKeyCol := attrColumnsMap[ValueTypeBool].keyCol
+	boolKeyCol.(*proto.ColArr[string]).Append(group.BoolKeys)
```

Comment:
I am not sure I follow the data ownership here. `AllAttributeColumns` is a static value. So are `attrColumnsMap` and `boolKeyCol`. What happens when we  we call `boolKeyCol.Append(group.BoolKeys)` where `group.BoolKeys` is a dynamic value for a given trace only?